### PR TITLE
add github actions ci workflow

### PR DESCRIPTION
**Pull request by mcfiredrill**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # Install dependencies
+      - name: Install dependencies
+        run: yarn install
```
Comment (mcfiredrill) [inline .github/workflows/ci.yml]: We could use npm if preferred?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Install the correct version of node
+      - name: Detect Node version
+        run: echo "##[set-output name=version;]$(cat .nvmrc)"
```
Comment (mcfiredrill) [inline .github/workflows/ci.yml:14]: I didn't see any engines key in the package.json, should we just use the latest node version?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (auggod) [reply .github/workflows/ci.yml]: We should not run npm install directly nor yarn. Currently, we use lerna, so the command `lerna bootstrap` should be all that we need once it's installed.

Comment (mcfiredrill) [reply .github/workflows/ci.yml]: Got it.
Testing this on my fork here, it fails with:
```
Run lerna bootstrap
  lerna bootstrap
  shell: /usr/bin/bash -e {0}
lerna notice cli v4.0.0
lerna info versioning independent
lerna info ci enabled
lerna info bootstrap root only
npm ERR! cipm can only install packages with an existing package-lock.json or npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or later to generate it, then try again.
```

I understand we don't have a package-lock.json due to the way lerna works, however....

Comment (mcfiredrill) [reply .github/workflows/ci.yml]: Got it working with the `--no-ci` flag https://github.com/mcfiredrill/stream/runs/3991608243?check_suite_focus=true

**Pull request merged**